### PR TITLE
test: add more abstract and reduce duplicate code

### PR DESCRIPTION
**Pull request by kruskall**
Removes duplicate code completely

Closes https://github.com/elastic/apm-queue/issues/151

### CODE REVIEW

**Review by dmathieu — COMMENTED** [review]
In the CI output, the logger seems to show all debug logs, when it seems it should only be errors?

**Comment by kruskall** [comment]
> In the CI output, the logger seems to show all debug logs, when it seems it should only be errors?

CI is not running the system tests yet. The test job is running the unit tests, some of which are using the logger with debug level (see graceful shutdown unit tests)

**Comment by dmathieu** [comment]
Ah, my bad. It's another test, with a similar name.

**Review by marclop — APPROVED** [review]
LGTM other than the `t.Skip()`.

File: systemtest/produce_consume_delivery_test.go [hunk 1]
```diff
@@ -27,134 +27,111 @@ import (
...
-			var errored atomic.Int32
-			cfg := kafka.ConsumerConfig{
-				CommonConfig:   kafka.CommonConfig{Logger: baseLogger},
-				Delivery:       tc.deliveryType,
-				Decoder:        codec,
-				Topics:         topics,
-				GroupID:        "groupid",
-				MaxPollRecords: tc.maxPollRecords,
-				Processor: model.ProcessBatchFunc(func(ctx context.Context, b *model.Batch) error {
+	t.Skip()
```
[the remark this answers is not in the document]
Comment (kruskall) [reply systemtest/produce_consume_delivery_test.go:36]: Yes, this test is flaky so I skipped it for now! I opened a followup issue to investigate

**Pull request merged**